strings/base_coroutine_threadpool.h

```diff
@@ -435,10 +435,10 @@ WINRT_EXPORT namespace winrt
 
             void fire_immediately() noexcept
             {
-                if (WINRT_IMPL_SetThreadpoolTimerEx(m_timer.get(), nullptr, 0, 0))
+                if (WINRT_IMPL_SetThreadpoolTimer(m_timer.get(), nullptr, 0, 0))
```

Comment:
According to [SetThreadpoolTimer docs](https://docs.microsoft.com/en-us/windows/win32/api/threadpoolapiset/nf-threadpoolapiset-setthreadpooltimer), it returns void instead of BOOL, is it OK to use this return value?

Reply:
Well spotted - @oldnewthing I'm guessing this means we don't have test coverage for this one. 😟

Reply:
Strange, because [there's a cancellation test](https://github.com/microsoft/cppwinrt/blob/1502e293d6e8f0a2c017431f6cf186d1551202fe/test/test/async_propagate_cancel.cpp#L76). Will have to figure out how this slipped through. Probably because I messed up.

Reply:
Okay, the deal is that this works on Win8+ due to how SetThreadpoolTimer is implemented (it calls SetThreadpoolTimerEx and throws away the result, but the result is still lying around in eax so we can still act on it). But this won't work on Win7, where eax does not contain anything meaningful. It looks like timer cancellation propagation will not work on Win7. Should I do GetProcAddress/dynamic light-up?

Reply:
Yes, we'll have to do that. You can (almost) use `load_runtime_function` except that its currently only used for combase functions and the thread pool is implemented by kernel32.

https://github.com/microsoft/cppwinrt/blob/1502e293d6e8f0a2c017431f6cf186d1551202fe/strings/base_agile_ref.h#L124-L140

This is the way Win7 fallback is handled, for example:

https://github.com/microsoft/cppwinrt/blob/1502e293d6e8f0a2c017431f6cf186d1551202fe/strings/base_activation.h#L21-L38

Reply:
Can't quite do it that way because I want to disable a feature if the GetProcAddress fails. There is no "fallback" available. (I guess I could see if result == fallback_Blah.)

Reply:
Whatever you think is best.

Reply:
#898